libs/cardano-ledger-conformance/src/Test/Cardano/Ledger/Conformance/ExecSpecRule/Conway/Deleg.hs

```diff
@@ -25,13 +27,26 @@ import Test.Cardano.Ledger.Conformance.SpecTranslate.Conway.Base ()
 import Test.Cardano.Ledger.Conformance.SpecTranslate.Conway.Cert ()
 import Test.Cardano.Ledger.Conformance.SpecTranslate.Conway.Deleg ()
 import Test.Cardano.Ledger.Constrained.Conway
+import Test.Cardano.Ledger.Constrained.Conway.WitnessUniverse
+
+instance
+  Inject
+    (WitUniv Conway, Set (Credential 'DRepRole StandardCrypto))
+    (Set (Credential 'DRepRole StandardCrypto))
+  where
+  inject (_, x) = x
 
 instance IsConwayUniv fn => ExecSpecRule fn "DELEG" Conway where
-  type ExecContext fn "DELEG" Conway = Set (Credential 'DRepRole StandardCrypto)
+  type ExecContext fn "DELEG" Conway = (WitUniv Conway, Set (Credential 'DRepRole StandardCrypto))
```

Comment:
I would create a new `ConwayDelegExecContext` datatype that contains the universe and the set of `DRep` credentials.

Reply:
same as before, the separation of concerns is important.